### ISSUE TITLE
Use appropriate caution when handling configurations that have complex rewrite logic

### ISSUE THREAD

**Issue by pde** [certbot/certbot#3252]
If the user has a config file with complex redirects (eg conditional redirects), it's in general impossible for us to add HTTP -> HTTPS redirects safely.  It's even hard to create new vhosts safely, which is necessary in Apache (cf #3210 ).  Where possible, we should try to detect the sorts of conditional redirecting logic that create this danger ([here's an nginx example](http://redant.com.au/ruby-on-rails-devops/manage-ssl-redirection-in-nginx-using-maps-and-save-the-universe/); [here's an Apache example](http://serverfault.com/a/609348); but of course these will come in many flavours).

If we detect this sort of conditionality, we should at minimum warn the user before enabling any redirects, or possibly refuse to do it without some sort of override.

**Comment by bmw** [comment]
After talking about this on the call, we determined the safest thing we can do here is just refuse to enable the redirect if there are already rewrite rules for the vhost. We proposed we could:

1. Add rewrite directives as a comment, telling the user they can uncomment these lines if they are confident it won't create a redirect loop.
2. Dump the directives required to the terminal (this might be a bit uglier when rewrites couldn't be enabled for multiple vhosts).
